plugin/src/test/java/com/microsoft/java/bs/gradle/plugin/GradleBuildServerPluginTest.java

```diff
@@ -41,28 +45,121 @@ static void beforeClass() {
+    File initScript = PluginHelper.getInitScript();
+    modelBuilder
+        .addArguments("--init-script", initScript.getAbsolutePath())
+        .addArguments("-Dorg.gradle.daemon.idletimeout=10")
+        .addArguments("-Dorg.gradle.vfs.watch=false")
+        .addArguments("-Dorg.gradle.logging.level=quiet")
+        .addJvmArguments("-Dbsp.gradle.supportedLanguages="
+            + String.join(",", SupportedLanguages.allBspNames));
+    return new DefaultGradleSourceSets(modelBuilder.get());
+  }
+
+  private interface ConnectionConsumer {
+    void accept(ProjectConnection connection) throws IOException;
+  }
+  
+  private void withConnection(File projectDir, GradleVersion gradleVersion,
+      ConnectionConsumer consumer) throws IOException {
+    GradleConnector connector = GradleConnector.newConnector()
+        .forProjectDirectory(projectDir)
+        .useGradleVersion(gradleVersion.getVersion());
     try (ProjectConnection connect = connector.connect()) {
+      consumer.accept(connect);
+    } finally {
+      connector.disconnect();
+    }
+  }
+
+  private void withSourceSets(String projectName, GradleVersion gradleVersion,
+      Consumer<GradleSourceSets> consumer) throws IOException {
+    File projectDir = projectPath.resolve(projectName).toFile();
+    withConnection(projectDir, gradleVersion, connect -> {
       GradleSourceSets gradleSourceSets = getGradleSourceSets(connect);
-      assertEquals(2, gradleSourceSets.getGradleSourceSets().size());
       for (GradleSourceSet gradleSourceSet : gradleSourceSets.getGradleSourceSets()) {
-        assertNotNull(gradleSourceSet.getGradleVersion());
-        assertEquals("junit5-jupiter-starter-gradle", gradleSourceSet.getProjectName());
-        assertEquals(":", gradleSourceSet.getProjectPath());
+        assertEquals(gradleVersion.getVersion(), gradleSourceSet.getGradleVersion());
+        assertEquals(projectName, gradleSourceSet.getProjectName());
         assertEquals(projectDir, gradleSourceSet.getProjectDir());
         assertEquals(projectDir, gradleSourceSet.getRootDir());
+      }
+      consumer.accept(gradleSourceSets);
+    });
+  }
+
+  private static int getJavaVersion() {
+    String version = System.getProperty("java.version");
+    if (version.startsWith("1.")) {
+      version = version.substring(2, 3);
+    } else {
+      int dot = version.indexOf(".");
+      if (dot != -1) {
+        version = version.substring(0, dot);
+      }
+    }
+    return Integer.parseInt(version);
+  }
+
+  private static class GradleJreVersion {
+    final String gradleVersion;
+    final int jreVersion;
+
+    GradleJreVersion(String gradleVersion, int jreVersion) {
+      this.gradleVersion = gradleVersion;
+      this.jreVersion = jreVersion;
+    }
+
+    GradleVersion getGradleVersion() {
+      return GradleVersion.version(gradleVersion);
+    }
+  }
+  
+  /**
+   * create a list of gradle versions that work with the runtime JRE.
```

Comment:
Would you mind adding some more comments explaining how the versions come from and how to add new one into the list? Basically the discussion in https://github.com/microsoft/build-server-for-gradle/pull/149#issuecomment-2159887903

Reply:
I've added comments per version